File: protocol/src/test/scala/com/zfoo/scala/packet/EmptyObject.scala
package com.zfoo.scala.packet
import com.zfoo.scala.IProtocolRegistration
import com.zfoo.scala.ByteBuffer
import scala.collection.mutable


class EmptyObject {
  
}

object RegistrationEmptyObject extends IProtocolRegistration {
  override def protocolId: Short = 0

  override def write(buffer: ByteBuffer, packet: Any): Unit = {
    if (packet == null) {
      buffer.writeInt(0)
      return
    }
    val message = packet.asInstanceOf[EmptyObject]
    buffer.writeInt(-1)
  }

  override def read(buffer: ByteBuffer): AnyRef = {
    val length: Int = buffer.readInt
    if (length == 0) return null
    val beforeReadIndex: Int = buffer.getReadOffset
    val packet: EmptyObject = new EmptyObject
    
    if (length > 0) buffer.setReadOffset(beforeReadIndex + length)
    packet
  }
}

File: protocol/src/test/kotlin/com/zfoo/kotlin/packet/EmptyObject.kt
package com.zfoo.kotlin.packet
import com.zfoo.kotlin.IProtocolRegistration
import com.zfoo.kotlin.ByteBuffer


class EmptyObject {
    
}

class EmptyObjectRegistration : IProtocolRegistration {
    override fun protocolId(): Short {
        return 0
    }

    override fun write(buffer: ByteBuffer, packet: Any?) {
        if (packet == null) {
            buffer.writeInt(0)
            return
        }
        val message = packet as EmptyObject
        buffer.writeInt(-1)
    }

    override fun read(buffer: ByteBuffer): Any {
        val length = buffer.readInt()
        val packet = EmptyObject()
        if (length == 0) {
            return packet
        }
        val beforeReadIndex = buffer.getReadOffset()
        
        if (length > 0) {
            buffer.setReadOffset(beforeReadIndex + length)
        }
        return packet
    }
}